testsuite/unittests/test_gap_junction.sli

```diff
@@ -204,5 +204,83 @@ M_ERROR setverbosity
   and
 } assert_or_die
 
+% Check that we can use gap junction with target-driven connect in topology
+{
+  ResetKernel
+  /lr << /rows 2
+         /columns 3
+         /elements /hh_psc_alpha_gap      
+      >> CreateLayer def
+  
+  lr lr << /connection_type (convergent)
+           /synapse_model /gap_junction
+           /make_symmetric true
+           /weights 0.5
+        >> ConnectLayers
+} pass_or_die
+
+% Check that we can use gap junction with source-driven connect in topology
+{
+  ResetKernel
+  /lr << /rows 2
+         /columns 3
+         /elements /hh_psc_alpha_gap      
+      >> CreateLayer def
+  
+  lr lr << /connection_type (divergent)
+           /synapse_model /gap_junction 
+           /make_symmetric true
+           /weights 0.5
+        >> ConnectLayers
+} pass_or_die
+
+% Check that we can use gap junction with convergent connect in topology
+{
+  ResetKernel
+  /lr << /rows 2
+         /columns 3
+         /elements /hh_psc_alpha_gap      
+      >> CreateLayer def
+  
+  lr lr << /connection_type (convergent)
+           /synapse_model /gap_junction 
+           /make_symmetric true
+           /weights 0.5
+           /number_of_connections 3
+        >> ConnectLayers
+} pass_or_die
+
+
+% Check that we can use gap junction with divergent connect in topology
+{
+  ResetKernel
+  /lr << /rows 2
+         /columns 3
+         /elements /hh_psc_alpha_gap      
+      >> CreateLayer def
+  
+  lr lr << /connection_type (divergent)
+           /synapse_model /gap_junction 
+           /make_symmetric true
+           /weights 0.5
+           /number_of_connections 3
+        >> ConnectLayers
+} pass_or_die
+
+% Check that non-symmetric gap junctions cannot be created in topology
+{
+  ResetKernel
+  /lr << /rows 2
+         /columns 3
+         /elements /hh_psc_alpha_gap      
+      >> CreateLayer def
+  
+  lr lr << /connection_type (convergent)
+           /synapse_model /gap_junction 
+           /make_symmetric false
+           /weights 0.5
+        >> ConnectLayers
+} fail_or_die
+
```

Comment:
The tests are fine, but i am wondering if we should add an additional test, that actually tests if the obtained results are valid. Maybe an mpi test similar to `testsuite/mpitests/test_gap_junctions_mpi.sli`?

Reply:
This test is too weak on several accounts: (i) It only tests that the code runs, not that the resulting connections are correct; (ii) in the first two cases it creates all-to-all connections, so one cannot check that forward and backward connections are symmetric; (iii) all connections have the same weight, so one cannot check whether forward and backward connection have identical weight; (iv) the comments for test cases 3 and 4 are the same as for 1 and 2, respectively.

And I agree with @janhahne that we should also have a test ensuring that correct symmetric connections are create when using MPI and threads.